chapters/classes.tex

```diff
@@ -278,35 +278,48 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
 
 \subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings-of-constants-and-parameters}
 
-The unexpanded binding equations for parameters and constants in the translated model must be acyclic after flattening; except that cycles are allowed if the cycles disappear when evaluating parameters having annotation \lstinline!Evaluate = true! that are not part of the cycle.
-Thus it is not possible to introduce equations for parameters by cyclic dependencies.
+For a constant or parameter \lstinline!v! with declaration equation, the unexpanded expression of the declaration equation in the flattended model must not depend on \lstinline!v! itself, neither directly nor indirectly via other variables' declaration equations.
```

Comment:
The term _unexpanded_ used here is something I'm not sure I understand what it means.  Is this parenthesis from the _Modelica DAE Representation_ appendix where it gets defined?
>   Expanding all class definitions (flattening the inheritance tree) and adding the equations and assignment statements of the expanded classes for every instance of the model.

If so, we should probably make that definition more prominent.  If not, can this rule be expressed using other (established) terms?  (@HansOlsson, I'm guessing that you are the most qualified person to answer this.)

Reply:
I think expand usually means expanding all records operator records, and arrays to a set of scalars of primitive types. (At least, this is how it appears in Section 4.7 Balanced Models. The HasCycles example has cycles when you look at the arrays. However, if you expand the equations into the equations for scalars there are no cycles left.

Reply:
Sounds like a plausible interpretation.  If no other interpretation pops up, I'll try to address this by either adding _expand_ to the index or adding a link link directly to where it is explained.

Reply:
Yes, unexpanded here was to ensure that we don't have to make a special case for x and y in:
```
final constant Real A [3 , 3] = [0 , 0 , 0; 1 , 0 , 0; 2 , 3 , 0];
parameter Real y [3] = A * y + ones (3) ;
parameter Real x [ n ] = cat (1 , {3.4} , x [1:( n -1) ]) ;
```
I'm not aware if there are any realistic use-cases for this. The benefit is both that we specifying the details (should some functions be inlined?), and that it doesn't require tools to scalarize equations needlessly to check this.

Reply:
Clarified in [commit](https://github.com/modelica/ModelicaSpecification/pull/2754/commits/b1b0dd4fe982e975720b40afdfd6e53ad1807c2b).  I didn't find any useful way of linking to the meaning of _expanded_, so I wrote it with more words instead, similar to the other places. OK?

Reply:
I take that as absence of objections.  Resolving.